modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/JacksonGenerator.scala

```diff
@@ -296,13 +300,55 @@ object JacksonGenerator {
         case _ if parents.length == 1                   => Target.pure(parents.headOption)
         case _                                          => Target.pure(None)
       }
+
+      discriminators                             = parents.flatMap(_.discriminators)
+      discriminatorNames                         = discriminators.map(_.propertyName).toSet
+      parentParams                               = parentOpt.toList.flatMap(_.params)
+      parentParamNames                           = parentParams.map(_.name)
+      (parentRequiredTerms, parentOptionalTerms) = sortParams(parentParams)
+      parentTerms                                = parentRequiredTerms ++ parentOptionalTerms
+
+      discriminatorValues <- parentTerms
+        .flatMap({ term =>
+          discriminators.find(_.propertyName == term.propertyName).map((term, _))
+        })
+        .traverse({
+          case (term, discriminator) =>
+            val discriminatorValue = discriminator.mapping
+              .collectFirst({ case (value, elem) if elem.name == clsName => value })
+              .getOrElse(clsName)
+
+            def parseLiteral[T](parser: String => T, friendlyName: String): Target[T] =
+              Try(parser(discriminatorValue)).fold(
+                t => Target.raiseError[T](s"Unable to parse '$discriminatorValue' as $friendlyName: ${t.getMessage}"),
+                Target.pure[T]
+              )
+
+            term.fieldType.simpleName
+              .fold(
+                Target.raiseError[Expression](s"Field type for discriminator ${term.propertyName} is not valid for this type")
+              )({
+                case "String"              => Target.pure[Expression](new StringLiteralExpr(discriminatorValue))
+                case "boolean" | "Boolean" => parseLiteral(_.toBoolean, "boolean").map(new BooleanLiteralExpr(_))
```

Comment:
Why is it possible to get both cases here? Unboxed vs boxed?

Reply:
Yep.